### DISCUSSION TITLE
Can I use prefect server without docker?

### DISCUSSION THREAD

**Discussion by Sinha-Ujjawal** [PrefectHQ/prefect#3731] [Q&A]
Can I use prefect server without docker?

**Comment by joshmeek** [comment]
Hi @Sinha-Ujjawal the server README has instructions to start Prefect Server directly without using the core CLI's docker-compose set up https://github.com/PrefectHQ/server#running-the-system-as-a-developer

**Reply by Sinha-Ujjawal** [reply]
It does not work in windows. Getting the following error-
```bash
module 'os' has no attribute 'setsid'
```
Seems like it will only work in Linux. Is it?

**Reply by joshmeek** [reply]
Apologies that I am just now getting around to responding to this! Hmm I do not have a windows machine and have not encountered this. Would you mind opening an issue report on the server repo?

**Comment by reubano** [comment]
@joshmeek Step 1 says `Start the database and Hasura in Docker:`. Is that a typo, or does it still require docker?

**Reply by joshmeek** [reply]
Ah running postgres and hasura locally is generally accomplished by running their respective docker images. If you want to avoid that all together you will have to start postgres and hasura manually which I don't think is covered in the server repo's quick-start README.

**Comment by SeimSoft** [comment]
Is there a solution? I would be really interested, since I cannot get the "docker in docker" setup to work.
I need to run it ideally inside a single docker without advanced privileges. I cannot get it to work...

**Reply by zanieb** [reply]
We do not support this. As noted previously, you can run our services outside docker as demonstrated in the Prefect Server development commands. Hasura and Postgres will need to be run independently following their documentation. 

I would not recommend going this route, the Server stack involves several services and Docker is the simplest way to orchestrate them. Prefect Orion will not require Docker to run the server.
